client/state/jetpack-connect/selectors.js

```diff
@@ -21,4 +21,17 @@ const getFlowType = function( state, site ) {
 	return false;
 };
 
-export default { isCalypsoStartedConnection, getFlowType };
+const hasXmlrpcError = function( state ) {
+	return (
+		state.authorizeError &&
+		state.authorizeError.message &&
+		state.authorizationCode &&
+		(
+			state.authorizeError.message.indexOf( 'transport error' ) > -1 ||
```

Comment:
Is it necessary to list all types of errors? Maybe just checking for the existence of `error` would work as fine?